server/src/main/java/org/opensearch/index/store/RemoteDirectory.java

```diff
@@ -149,7 +185,7 @@ public IndexInput openInput(String name, IOContext context) throws IOException {
         InputStream inputStream = null;
         try {
             inputStream = blobContainer.readBlob(name);
-            return new RemoteIndexInput(name, inputStream, fileLength(name));
+            return new RemoteIndexInput(name, downloadRateLimiter.apply(inputStream), fileLength(name));
```

Comment:
this is applying rate limited after reading from blobContainer . We will need to apply it before.

Reply:
No it's acting on the underlying input stream